### ISSUE TITLE
Determining whether a host folder is mounted is not taking source and target into account

### ISSUE THREAD

**Issue by hferentschik** [minishift/minishift#2122]
The code looks like this:

```Go
	procMounts := ""
	if isRunning {
		cmd := fmt.Sprint("cat /proc/mounts")
		procMounts, _ = drivers.RunSSHCommandFromDriver(driver, cmd)
	}

	hostfolders := miniConfig.AllInstancesConfig.HostFolders
	hostfolders = append(hostfolders, miniConfig.InstanceConfig.HostFolders...)
	var mounts []MountInfo
	for _, hostFolder := range hostfolders {

		source := ""
		switch hostFolder.Type {
		case CIFS.String():
			source = hostFolder.Options[config.UncPath]
		case SSHFS.String():
			source = hostFolder.Options[config.Source]
		}

		mounted := false
		if isRunning && strings.Contains(procMounts, hostFolder.MountPoint()) {
			mounted = true
		}

		mount := MountInfo{
			Name:       hostFolder.Name,
			Type:       hostFolder.Type,
			Source:     source,
			MountPoint: hostFolder.MountPoint(),
			Mounted:    mounted,
		}

		mounts = append(mounts, mount)
	}
```

The code only checks whether the output of `cat /proc/mounts` contains the mount point. However, this fails, if for example the mountpoints are foo and foo1. Really the output should be considered line by line and take hostfolder source and target into account.

**Comment by gbraad** [comment]
The above code can be found here:

https://github.com/minishift/minishift/blob/c56f051fa872e8fd15d41544f6096e01f8a6845a/pkg/minishift/hostfolder/hostfolder_manager.go#L103-L136

We might also have to check:

https://github.com/minishift/minishift/blob/c56f051fa872e8fd15d41544f6096e01f8a6845a/pkg/minishift/hostfolder/hostfolder_manager.go#L284-L297

**Comment by LalatenduMohanty** [comment]
> The code only checks whether the output of cat /proc/mounts contains the mount point. However, this fails, if for example the mountpoints are foo and foo1. Really the output should be considered line by line and take hostfolder source and target into account.

+1 . We need to fix it soon as users will find this as a bug.

**Comment by LalatenduMohanty** [comment]
Resolved via #2240
